doc/interoperability/pandas.rst

```diff
@@ -0,0 +1,295 @@
+
+.. _pandas:
+
+=======================
+Pandas Interoperability
+=======================
+
+This part of the User Guide aims to highlight some of the particularities that
+arise when using pandas datastructures as input for scikit-learn.
+
+The basics of using pandas and Scikit-learn
+==================================================================
+
+Scikit-learn supports the use of
+`pandas DataFrames <http://pandas.pydata.org/pandas-docs/stable/>`__
+implicitly. However, this implicit support has its conditions and potential
+pitfalls and some (not all) of these will be outlined below.
+
+Reasons for these conditions and the difficulty of supporting pandas, stems
+from the fact that pandas data structures such as Series and DataFrames can
+hold heterogenous datatypes (different columns can contain different
+datatypes). Therefore, Series and DataFrames can be thought of as containers
+for arrays that hold the actual data in an homogenous format. Some of these
+underlying data structures might not necessarily be representable as NumPy
+arrays (e.g. Categorical data). Pandas supports these alternate data types using
+pandas specific or 3rd party libraries that extend the NumPy type system. The
+difficulty of supporting pandas data structure thus comes from pandas
+capability of supporting heterogenous data in one data structure.
+
+Every Scikit-learn estimator/transformer/pipeline
+(for the rest of this section we shall call these primitives)
+supports the use of DataFrames as inputs which is achieved by obtaining a
+`NumPy array <https://docs.scipy.org/doc/numpy/user/>`__ using
+the :meth:`~numpy.asarray` on a DataFrame object. The only exception where a
```

Comment:
Not the only exception. Several meta-estimators pass their data on unmodified (Pipeline, etc.) or know how to slice a Pandas DataFrame (GridSearchCV, etc.)